cms/apps/media/admin.py

```diff
@@ -159,6 +161,88 @@ def get_title(self, obj):
     get_title.admin_order_field = 'title'
     get_title.short_description = 'title'
 
+    def get_related_pages(self, querysets):
+        pages = []
+        for queryset in querysets:
+            for model in queryset:
+                if hasattr(model, 'page'):
+                    pages.append(model)
+                else:
+                    pages.append(model.__class__.objects.get(pk=model.id))
```

Comment:
Correct me if I am wrong, but this looks like the if-clause does the same thing in both cases. In the second clause you're getting the instance of that object's class with the PK of the object's class? But that seems to me that it's exactly the same thing as `model`.

Reply:
You appear to be correct. I've now removed the if clause and just used model (now renamed to obj).